code/modules/lost_crew/damages/_damages.dm

```diff
@@ -0,0 +1,107 @@
+
+/// Generate and apply a possible character (species etc)
+/datum/corpse_damage_class/proc/apply_character(mob/living/carbon/human/fashion_corpse, list/protected_objects, list/datum/callback/on_revive_and_player_occupancy, list/body_data)
+	var/datum/corpse_character/character = pick_weight(possible_character_types)
+	character = new character()
+	character.apply_character(fashion_corpse, protected_objects, on_revive_and_player_occupancy)
+
+	var/datum/corpse_assignment/assignment = pick_weight(possible_character_assignments)
+	if(assignment)
+		assignment = new assignment()
+		assignment.apply_assignment(fashion_corpse, protected_objects, on_revive_and_player_occupancy)
+
+	death_lore += assignment?.job_lore
+
+	body_data += character.type
+	body_data += assignment?.type
+
+/// Set up injuries
+/datum/corpse_damage_class/proc/apply_injuries(mob/living/carbon/human/victim, list/saved_objects, list/datum/callback/on_revive_and_player_occupancy, list/body_data)
+	var/datum/corpse_damage/cause_of_death/cause_of_death = pick_damage_type(possible_causes_of_death)
+	cause_of_death.apply_to_body(victim, rand(), saved_objects, on_revive_and_player_occupancy)
+
+	var/datum/corpse_damage/post_mortem = pick_damage_type(post_mortem_effects)
+	post_mortem?.apply_to_body(victim, rand(), saved_objects, on_revive_and_player_occupancy)
+
+	var/datum/corpse_damage/decay = pick_damage_type(decays)
+	decay?.apply_to_body(victim, rand(), saved_objects, on_revive_and_player_occupancy)
+
+	// Simulate bloodloss by dragging/moving
+	victim.blood_volume = max(victim.blood_volume - victim.bleedDragAmount() * rand(20, 100), 0)
+	set_death_date(victim)
+
+	body_data += cause_of_death.type
+	body_data += post_mortem?.type
+	body_data += decay.type
+
+	death_lore += area_lore + " " + cause_of_death.cause_of_death
+
+/// Wrapped pickweight so we can have a bit more controle over how we pick our rules
+/datum/corpse_damage_class/proc/pick_damage_type(list/damages, list/used_damage_types)
+	var/list/possible_damages = list()
+
+	for(var/datum/corpse_damage/damage as anything in damages)
+		if(isdatum(damage) && damage.damage_type && (damage.damage_type in used_damage_types))
+			continue
+		possible_damages[damage] = damages[damage]
+
+	var/datum/corpse_damage/chosen = pick_weight(possible_damages)
+	if(!ispath(chosen)) //can also be null for some variants
+		return null
+
+	chosen = new chosen()
+
+	if(chosen.damage_type)
+		used_damage_types += chosen.damage_type
+	return chosen
+
+/// Soulfully give a date of death for health analyzers
+/datum/corpse_damage_class/proc/set_death_date(mob/living/carbon/body)
+	var/current_space_time = world.realtime + 174709440000 //254 years in deciseconds. byond breaks if you do it manually :/
+	var/died_how_long_ago = rand(lore_death_time_min, lore_death_time_max)
+	body.timeofdeath = current_space_time - died_how_long_ago
+	body.station_timestamp_timeofdeath = time2text(body.timeofdeath, "DD Month YYYY")
```

Comment:
Seems like this is causing it to like, get some big weirdass `died -9.57546e+08 seconds ago` type output on the autopsy analyzer dead body health scan?
I imagine it's because regular death uses `world.realtime` directly:
https://github.com/tgstation/tgstation/blob/dbe110d4d09780586bd71cbcd6e3e8409256ac07/code/modules/mob/living/death.dm#L118-L119
While we here use `world.realtime + 174709440000` as the basis, which may be registering as them having died in the future.

Also seems like it's mostly setting them as like having died in years in the 2000s?

Reply:
Yeah the numbers are too big and it causes an overflow :/ . I’ve already spent some time debugging it and it seems pretty hopeless. I’d rather keep this in to indicate a long time has passed and the analyzer is struggling, than it showing they died 1 second ago.

Proper fix would be to construct the time and dates manually and change all code relying on it, but I wont have time (or energy) to do that for quite a while (and this PR is already too big for my taste)

Reply:
cool cool o7
took a bit to look into it myself in your stead

Seems the `timeofdeath` part was wonky due to it using `world.realtime` while the underlying system uses `world.time`.
On the other end, we can avoid the timestamp issue by just calculating it based on current `world.realtime`, converting the resulting year to a number much like we do elsewhere, and adding in the `STATION_YEAR_OFFSET` _after_ we calculate it.
This seems to do the job:
![image](https://github.com/user-attachments/assets/d8b77799-d1f9-465b-adef-9e2572bacb09)
![image](https://github.com/user-attachments/assets/d1d3eadd-12a5-4946-941b-3a991c2b9b38)
![image](https://github.com/user-attachments/assets/8e72256d-6976-433d-ba1d-aa9273fdd73d)

```suggestion
/// Soulfully give a date of death for health analyzers
/datum/corpse_damage_class/proc/set_death_date(mob/living/carbon/body)
	var/died_how_long_ago = rand(lore_death_time_min, lore_death_time_max)
	body.timeofdeath = world.time - died_how_long_ago

	var/death_real_time = world.realtime - died_how_long_ago
	var/current_date = time2text(death_real_time, "DD Month")
	var/current_year = text2num(time2text(death_real_time, "YYYY")) + STATION_YEAR_OFFSET
	body.station_timestamp_timeofdeath = "[current_date] [current_year]"
```
the code in question if ya feel like it

Reply:
displaying "1704 days" or whatever for longer than a year is a bit wonk, but like, I feel there's no need to try shove that on top of here too with how big this already is

Reply:
you're actually insane, solid beyond solid